src/test/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerGenerateContractPartyCsvServiceImplTest.java

```diff
@@ -187,5 +212,151 @@ static Stream<Arguments> jaggaerContractPartyUploadDtoToStringParameters() {
+        dto.setName(VENDOR3_ADDRESS1_NAME);
+        dto.setAddressType(JaggaerAddressType.FULFILLMENT);
+        dto.setPrimaryType(VENDOR3_ADDRESS1_PRIMARY_TYPE);
+        dto.setActive(VENDOR3_ACTIVE);
+        dto.setCountry(COUNTRY_US);
+        dto.setStreetLine1(VENDOR3_ADDRESS1_STREET_LINE_1);
+        dto.setStreetLine2(VENDOR3_ADDRESS1_STREET_LINE_2);
+        dto.setStreetLine3(VENDOR3_ADDRESS1_STREET_LINE_3);
+        dto.setCity(VENDOR3_ADDRESS1_CITY);
+        dto.setState(STATE_NY);
+        dto.setPostalCode(VENDOR3_ADDRESS1_ZIP);
+        dto.setPhone(VENDOR3_ADDRESS1_PHONE);
+        dto.setTollFreeNumber(VENDOR3_ADDRESS1_TOLL_FREE_NUMBER);
+        dto.setFax(VENDOR3_ADDRESS1_FAX);
+        dto.setNotes(notes);
+        return dto;
+    }
+
+    static String[] buildJaggaerContractAddressUploadDtoData(String notes) {
+        String[] record = { JaggaerContractPartyUploadRowType.ADDRESS.rowType, VENDOR3_ADDRESS1_ADDRESS_ID,
+                VENDOR3_ADDRESS1_SCIQUEST_ID, VENDOR3_ADDRESS1_NAME, JaggaerAddressType.FULFILLMENT.jaggaerAddressType,
+                VENDOR3_ADDRESS1_PRIMARY_TYPE, VENDOR3_ACTIVE, COUNTRY_US, VENDOR3_ADDRESS1_STREET_LINE_1,
+                VENDOR3_ADDRESS1_STREET_LINE_2, VENDOR3_ADDRESS1_STREET_LINE_3, VENDOR3_ADDRESS1_CITY, STATE_NY,
+                VENDOR3_ADDRESS1_ZIP, VENDOR3_ADDRESS1_PHONE, VENDOR3_ADDRESS1_TOLL_FREE_NUMBER, VENDOR3_ADDRESS1_FAX,
+                notes };
+        return record;
+    }
+    
+    @Test
+    void testBuildPartyHeader() {
+        String expectedPartyHeader = "PARTY,OverrideDupError,ERPNumber,SciQuestID,ContractPartyName,DoingBusinessAs,OtherNames,CountryOfOrigin,Active,ContractPartyType,Primary,LegalStructure,TaxIDType,TaxIdentificationNumber,VATRegistrationNumber,WebsiteURL";
+        String actualPartyHeader = printArrayCommaDelim(jaggaerGenerateContractPartyCsvServiceImpl.buildPartyHeader());
+        assertEquals(expectedPartyHeader, actualPartyHeader);
+    }
+
+    @Test
+    void testBuildAddressHeader() {
+        String expectedAddressHeader = "ADDRESS,AddressID,SciQuestID,Name,AddressType,PrimaryType,Active,Country,StreetLine1,StreetLine2,StreetLine3,City/Town,State/Province,PostalCode,Phone,TollFreeNumber,Fax,Notes";
+        String actualAddressHeader = printArrayCommaDelim(jaggaerGenerateContractPartyCsvServiceImpl.buildAddressHeader());
+        assertEquals(expectedAddressHeader, actualAddressHeader);
+    }
+
+    @Test
+    void testBuildContactHeader() {
+        String expectedContactHeader = "CONTACT,ContactID,SciQuestID,Name,FirstName,LastName,ContactType,PrimaryType,Active,Title,Email,Phone,MobilePhone,TollFreeNumber,Fax,Notes";
+        String actualContactHeader = printArrayCommaDelim(jaggaerGenerateContractPartyCsvServiceImpl.buildContactHeader());
+        assertEquals(expectedContactHeader, actualContactHeader);
+    }
+
+    private String printArrayCommaDelim(String[] data) {
+        StringBuilder sb = new StringBuilder();
```

Comment:
You could simplify this concatenation by using one of the StringUtils.join() methods, or by using a Stream and using Collectors.joining() to build the concatenated string.

Reply:
ok